Tests/sbgTests/FoundationFileAdderTests.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  FoundationFileAdderTests.swift
+//  SBGTests
+//
+//  Created by Dawid Markowski on 15/10/2018.
+//
+
+
+import Foundation
+import Quick
+import Nimble
+@testable import SBGCore
+
+class FoundationFileAdderTests: QuickSpec {
+    
+    override func spec() {
+        describe("FoundationFileAdder") {
+            
+            var sut: FoundationFileAdder!
+            var pathResolver: MockPathResolver!
+            
+            beforeEach {
+                pathResolver = MockPathResolver()
+                sut = FoundationFileAdder(pathResolver: pathResolver)
+            }
+            
+            context("when adding file") {
+                beforeEach {
+                    try! sut.addFile(with: MockConstants.sampleName, content: MockConstants.sampleContent, to: MockConstants.sampleDirectory)
+                }
+                
+                it("should invoke path resolver excactly once") {
```

Comment:
Minor: this is just a matter of preferences, but all `it` sections from this tests can be combined into a single `it("adds file")` (which verifies parameters and the number of invocations). 
Also, it would be good to verify if it throws an error on a failure.

Reply:
I like the rule that one test contains only one expectation, hence we started to use that convention. I would like to keep it if you don't mind.